### DISCUSSION TITLE
Please, please support excluding archived repos!

### DISCUSSION THREAD

**Discussion by serhalp** [community/community#8591] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi! I'm so excited that GitHub is rebuilding and significantly improving code search! It's desperately needed.

Please, please don't rebuild a brand new code search that **_also_ does not support filtering out results from archived repos**.

This missing functionality alone has easily made GitHub code search 50+% less usable than it would otherwise be for me at our org, which contains 900+ repos out of which 600+ are archived. I have to wade through so much noise while holding the context in my head about which repos are archived and therefore irrelevant. I have to let others know when they've been misled into thinking some function is used 25 times when it's actually used 3 times.

There are have been hundreds of people advocating for this for years over here: https://github.community/t/feature-request-search-code-for-not-archived-repositories/2135/56.

If I had children, I would give up my first-born son for this. 🙏🏼

**Comment by Avivhdr** [comment]
I don't have children but I would give someone else's child for it 👍🏼

**Reply by ericLemanissier** [reply]
I have two children, but I would give 10 other children for this feature (at least)

**Reply by vronchin** [reply]
I don't have children but I would make children and give half of the total for this feature (at least)

**Comment by jbarber** [comment]
I agree this functionality would be killer. I use [Sourcegraph](https://sourcegraph.com/) for this exact feature.

**Comment by reevesbr** [comment]
We have thousands of repos across our organization, being able to code search search active, non-archived repos (the recently announced log4j vulnerability is the trigger for me landing here) is a **must have**.

**Comment by look** [comment]
Thanks for the feedback! This is on our roadmap. (The team requests please no children, though.)

**Reply by serhalp** [reply]
@look That's great news! For an added push... it was on our roadmap to jump ship to Sourcegraph for search in the coming year. With the GitHub code search rebuild announcement, we're close to abandoning that project – but without this feature (excluding archived repos from code search), we'll have to proceed anyway; with it, we'll definitely not bother with an external tool (sorry, any Sourcegraph folks reading this, you seem great).

**Reply by devinschumacher** [reply]
still not done huh?

**Comment by wagnerbsouza** [comment]
That latest log4j vulnerability just brought me here. The archived repos in the search results are "annoying".

**Comment by look** [comment, marked as answer]
This is now implemented!

By default, results from archived repositories are searched, but down-ranked. If a result appears only in an archived repository, you'll see it.

You can change this using `is:archived` and `NOT is:archived`. For example, if you want to find Log4J in non-archived repositories only, use:

[log4j NOT is:archived](https://cs.github.com/?q=log4j%20NOT%20is%3Aarchived)

More details are available on [the syntax page](https://cs.github.com/about/syntax#is).

**Reply by jetersen** [reply]
@look Thanks for listening and implementing this and so quickly, LOVE IT! 😍

Time to go searching again. 👏👏👏👏

**Reply by Avivhdr** [reply]
Thank you and the Github code-search team 👏🏼

**Reply by klhex** [reply]
When will this feature be available on GitHub Enterprise (from which version on will it be included)?

**Reply by katylava** [reply]
~~Is this working for anyone? It's not working for anyone in my org. I'm trying a search with `org:oreillymedia surrogate-key NOT is:archived` and even the second result is from an archived repo.~~ I just realized this is for a new code search product that's not live yet, not the existing one. 😭

**Reply by drewboswell** [reply]
When will us poor corpos get this in Github Enterprise? :crying_cat_face:

**Reply by MelissaSnell** [reply]
Just doesn't seem to work for me. Would be really useful if it did.

**Reply by danmoorenhs** [reply]
yeah this should not be marked as answered if it's a feature we don't have :/

**Reply by look** [reply]
@MelissaSnell @danmoorenhs are you using the [new GitHub code search](https://github.com/features/code-search)? The feature exists there; if it's not working for you, please give us some example queries.

**Reply by danmoorenhs** [reply]
i can't access feature i applied for beta but am on a wait list :(  thanks for asking though :sniff:

**Reply by sami1250** [reply]
[السيرة الذاتية.PDF](https://github.com/community/community/files/10832600/default.PDF)

**Reply by deiga** [reply]
@look I'm using the new Search and this filter is not working currently
Search query: https://github.com/search?q=org%3Aamboss-mededu+label+path%3AJenkinsfile+NOT+is%3Aarchived&type=code
Results include https://github.com/amboss-mededu/dlp which is archived

**Reply by rubengmurray** [reply]
The query above seems to work for me: @deiga your link returns 0 results, but removing the `NOT is:archived` clause returns many results.

**Reply by deiga** [reply]
@rubengmurray Weird, I still have a file from the archived repo as the 7th result out of 34

**Reply by rubengmurray** [reply]
I wonder if the results are cached (cached by user search) :/? 

I'm getting behaviour similar to you with my organisation's repositories that I'm currently archiving; I archived 7 repos about 2 hours ago and all of them are still showing in search despite using `NOT is:archived` (and the ordering is also still the same).

Frustrating.

**Reply by rubengmurray** [reply]
Delays potentially related to: https://github.com/orgs/community/discussions/8591#discussioncomment-2815198, but there are also obvious complaints about this functionality not working beyond a search index.

@colinwm is there any guidance on what the actual search index interval is? It would be good to know so that indeed if that is the cause of the delay I'm seeing I'm not just intermittently checking ad infinitum until it may or may not appear.

Thanks

**Reply by look** [reply]
@rubengmurray thank you, this is useful context. I'm going to look into how we are handling archived state transitions.

We are currently rebuilding the index from scratch several times a month, so the correct state will eventually be reflected, but we would like it to be reflected soon after it happens.

**Reply by rubengmurray** [reply]
Thanks for the acknowledgement @look 

For the record, I stumbled on an org repository I archived back in August 2022 and that is being correctly omitted from search when `NOT is:archived` is used 👍

**Reply by rubengmurray** [reply]
Quick update: almost 24 hours later now and not seeing any difference. I'll take another look after 7 days and post an update.

**Reply by rubengmurray** [reply]
Still getting archived repositories in my search results as of today even though they were archived on 15th March (7+ days ago)...

**Reply by rubengmurray** [reply]
As of today all repositories archived on the 15th of March have disappeared from our search results. However I haven't checked this during the intervening period so my best estimate is between 7 and 28 days for this functionality to take effect.

**Reply by timotta** [reply]
Its not working on team repository search :(

**Reply by look** [reply]
@timotta I'm not totally sure what you mean by "team repository search", but the ability to exclude archived repositories we're discussing here is in code search.

**Reply by proehlen** [reply]
> You can change this using `is:archived` and `NOT is:archived`. 

I appreciate this is available now so, thanks, but I have to type that in for every search I do because I never want to see results in archived repositories.  It would be preferable if this was the default behavior.

**Reply by pascal-hofmann** [reply]
@look I noticed that this does not work on GitHub Enterprise Server. Is there a timeline on when we can expect this feature there?

**Reply by look** [reply]
@pascal-hofmann No timeline yet, but we are working on bringing the new code search to GitHub Enterprise Server right now.

**Reply by tejavegesna-lb** [reply]
You can use `archived:false` which works and show the repos

**Reply by AndreBlumenthal** [reply]
To be able to use `NOT is:archived` is a nice feature! One question to verify: I still see archived repositories in the results. All of those are archived "recently" - not more than two month ago. Is my observation correct, that there is a limitation to that feature?

**Reply by pklapperich** [reply]
While the [syntax page](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#is) describes using `is:archived` and `NOT is:archived`, when I try this on my organizations repository list it tells me that `archived` is not valid for `is:` and also it tells me to stop using `is:`

It looks like the accepted values for `is` are public, private, internal, and sponsorable.

![image](https://github.com/community/community/assets/12666157/541972d3-508b-445c-b1f4-77b13fb57cbe)


When I use the main search-box at the top of the page and do `org:myorganization is:archived` I get 141 repositories, some of them are archived and most aren't. When I do a simple search for `org:myorganization` I get the same list of 141 repositories.

`is:archived` doesn't seem to work. And when I search with `NOT is:archived` I just get no results.

**Reply by pklapperich** [reply]
> You can use `archived:false` which works and show the repos

This works, but it's not documented on the [syntax page](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax)

**Reply by devops-cmp** [reply]
Does it also work when using the api?

**Reply by thoraage** [reply]
`is:archived` works perfectly fine, but `NOT is:archived` does not work; I see both archived and non-archived. Frankly not seeing the archived repositories should be the default.

**Reply by mikhailvartanyan** [reply]
> `is:archived` works perfectly fine, but `NOT is:archived` does not work; I see both archived and non-archived. Frankly not seeing the archived repositories should be the default.

+1

**Reply by mbargull** [reply]
@thoraage, @mikhailvartanyan, `-is:archived` (i.e., negate by prefixing hyphen-minus) works for me.

**Reply by mikelindsey-okta** [reply]
This is not working with `-is:archived` nor with `NOT is:archived` ..  I have a search with 340 results without the exclusion.  When I add the exclusion I have 356 hits

**Reply by dreusel** [reply]
Also does not work for me, and there is no mention (anymore?) on the syntax pages.

**Reply by dreusel** [reply]
> Also does not work for me, and there is no mention (anymore?) on the syntax pages.

Ah, it actually does work, it just seems to take some time after archiving a repo until it actually gets excluded from search results.

**Reply by dev-willis** [reply]
I can't get `is:archived` or `NOT is:archived` to work at all but `archived:false/true` works.

**Reply by Gabriel-p** [reply]
Can confirm that to date `archived:false/true` works and the previous methods do not.

**Reply by roy09** [reply]
archived:false/true did not work for me.
what worked -> `is:archived` / `NOT is:archived`

**Reply by alebrozzo** [reply]
`archived:false` only works for issues for me, but not for actual code
`keyword NOT is:archived` returns a parsing error (the offending word being NOT)
`NOT is:archived keyword` returns more results than `keyword` with no other filter

I'm on Enterprise btw (it seems there may be a difference)

**Reply by sparrowt** [reply]
On GitHub Enterprise Server (GHES) 3.17 I find that `archived:false` and `NOT is:archived` both do _not_ work on the `Code` tab of search results, although they do work when searching for repositories (but that only searches things like the repo's name, not its content).

Interestingly both of these documentation pages for GHES:
https://docs.github.com/en/enterprise-server@3.17/search-github/searching-on-github/searching-code
https://docs.github.com/en/enterprise-server@3.18/search-github/searching-on-github/searching-code
say that:
> Archived repositories are not searchable.

which does not appear to match with my experience on GHES 3.17 where results from long-ago-archived repos still show up (regardless of any attempts at excluding them using either of the methods mentioned above).

On public github.com - and perhaps also GitHub Enterprise Cloud (GHEC)? - it seems that there are now 2 kinds of code search:
- "legacy" code search ([docs](https://docs.github.com/en/search-github/searching-on-github/searching-code)) which says _"Archived repositories are not searchable"_ (just like the GHES docs)
- "GitHub Code Search" which claims to support `NOT is:archived` for code too ([docs](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#is-qualifier))

Perhaps we can deduce therefore that the newer (non-legacy) "GitHub Code Search" is not available yet in on-prem GHES? See also [here](https://github.com/cli/cli/issues/10316#issuecomment-2825234261) which confirms the above re 2 implementations of code search in modern GH.

**Reply by SilverPreeceSOF** [reply]
December 2025 here, and the functionality is completely bugged. It works fine when searching for code, but when searching for repositories specifically - the _thing that can be archived_ - the behaviour is completely inverted. `is:archived` returns all my regular repositories, and `NOT is:archived` returns all my archived repositories. Judging from the replies above this is not a new issue. Feels to me like yet more evidence that since the big push toward All Copilot, All the Time at Microsoft, every product that isn't Copilot has been left to rot.

**Reply by jshinn-bip** [reply]
Is there any kind of delay between archiving a repo and this behavior being able to operate on that fact in the search?  Because I just archived a repo and searched with every mentioned iteration of the terms needed to exclude it and it did literally nothing.

**Comment by LivingInSyn** [comment]
I found `NOT is:archived` in the documentation for the current codesearch beta, but it doesn't work at all for me. Sample query I'm using is:

`NOT is:archived foo` 

returns archived repos

**Reply by colinwm** [reply]
Can you give a link to a more specific query? When I search `NOT is:archived foo`, I only get popular, non-archived repos:

https://cs.github.com/?scopeName=All+repos&scope=&q=NOT+is%3Aarchived+foo

**Reply by LivingInSyn** [reply]
an example:

```
NOT is:archived org:puppetlabs c_rehash
```

returns `https://github.com/puppetlabs/homebrew-core` in the results, which is an old archived repo

**Reply by colinwm** [reply]
I don't see it: https://cs.github.com/?scopeName=All+repos&scope=&q=NOT+is%3Aarchived+org%3Apuppetlabs+c_rehash

Are you using cs.github.com? Or github.com? This feature only works on cs.github.com (for now!)

**Reply by LivingInSyn** [reply]
this is on `cs.github.com`

<img width="959" alt="image" src="https://user-images.githubusercontent.com/6341769/170132438-1603013b-c1bc-487e-b3d7-71a862675423.png">

fwiw: I get different results logged in as an admin on the org vs not ( besides just seeing private repos)

**Reply by colinwm** [reply]
Did you recently archive this repo? It may take some time (~hours) for the archive status to reach the search index.

**Reply by LivingInSyn** [reply]
at least a week ago, likely much longer ago than that. This is just one example, there are many

**Reply by nkojuharov-aca** [reply]
I see the same - NOT is:archived is completely ignored - results for archived repos are shown near the top, completely obscuring results for non-archived repos.

**Reply by look** [reply]
@nkojuharov-aca sorry, we are clearly missing some updates when repos change archived state. We've prioritized a new approach to updating the index when repositories change to address the problems.

**Reply by dev-willis** [reply]
I had the same problem. I figured out that `archived:false` is what's needed.

**Comment by chrisrusso** [comment]
Please help, I have removed my github organization "codefen", and I need it back. I can't create a new one, the system says the name cannot be taken (probably because still saved somewhere can be recovered). Please help me!!!

**Reply by chrisrusso** [reply]
@github-staff please help!!  I have removed my github organization "codefen", and I need it back. I can't create a new one, the system says the name cannot be taken (probably because still saved somewhere can be recovered). Please help me!!!

**Reply by look** [reply]
@chrisrusso this is a feedback forum for Code Search and Code Navigation. Please contact customer support: https://support.github.com/contact?tags=rr-restore

**Comment by chrisrusso** [comment]
I think I erased it sometime between the last 3 weeks...

**Comment by chrisrusso** [comment]
> @chrisrusso this is a feedback forum for Code Search and Code Navigation. Please contact customer support: https://support.github.com/contact?tags=rr-restore

Thanks bro! I just contacted them! appreciated!!

**Comment by drullo** [comment]
This is only partially working for me.  When searching my organization's repos, adding 'NOT is:archived' to the search does cut down the number of results, but it unfortunately still includes some archived repos.

**Comment by brett-swan-sh** [comment]
I've found in our organization that this doesn't work at all. In fact, when you codesearch for something that _only exists in an archived repo_, and set `is: archived`, the code from the archived repo _doesn't show up_. But when you search using `NOT is:archived`, the code in the _archived_ repo does show up...

**Reply by look** [reply]
Sorry about that. This sounds like a data synchronization issue, which should be resolved the next time we rebuild the index. If you would like us to check on that, let me know the org name.

**Reply by brett-swan-sh** [reply]
Ah interesting, so there's a delay between archiving the repo and the search function reflecting archive status? Any idea what kind of delay we can expect for that?

**Reply by look** [reply]
There is a delay between a change happening and the search engine being updated, but it is also possible for a change to be processed out of order, which results in the archived status reflecting the old value. We correct this the next time a change happens, but for archived repositories, that's not usually until the next index rebuild.

**Reply by brett-swan-sh** [reply]
fair enough. Any idea what that delay looks like? 5, 15, 30 mins, 1 hour, 1 day?

**Reply by look** [reply]
The normal indexing delay is about 5 minutes. If we miss an update (like I suspect happened here), the fix won't roll out until the next rebuild. We do that about once a week or so.

**Comment by devinschumacher** [comment]
[Workaround]

Here's my workaround for all you frustrated cats (i was too till i just figured this out)

Create a new organization called `yourorganization-archive`

and instead of "archiving" a repo (i guess you still can do that too) you just "transfer" the repo.

boom! solved & saved :)

🍻 cheers!

now lets connect on social bc i need more developers and smart people feeds to read instead of political nonsense and camgirl spam

twitter [@dvnschmhr](https://serp.ly/@devin/twitter)
YT: [@devinschumacher](https://serp.ly/@devin/youtube)

**Reply by bitti** [reply]
One problem solved and more problems created: who will fix all the broken links to these 'archived' repos?

**Comment by rzuckerm** [comment]
When is this feature going to be rolled out to GitHub Enterprise? It looks like it's been in public GitHub for well over a year. I tried this on our 3.11.2 instance, and it does not work. For example:
```
filename:foo
```
returns N matches.
```
filename:foo is:archived
```
returns the same N matches, even though I know there are archived repos.
```
filename:foo NOT is:archived
```
returns no matches.

**Comment by stormindia** [comment]
If anyone can provide an update if this has been rolled out for GitHub (or for Github enterprise in particular)? Much like the original thread creator, I would go to have a child and give it to this cause (if it serves as an inspiration 🥺 )

**Reply by LiteBrite82** [reply]
Hi @stormindia 

Excellent question! Unfortunately, the answer is outside of my wheel house. 

The last update from @look (July 2023):
@pascal-hofmann asked:
> ... I noticed that this does not work on GitHub Enterprise Server. Is there a timeline on when we can expect this feature there?

look replied: 

> ... No timeline yet, but we are working on bringing the new code search to GitHub Enterprise Server right now.

I would, however, recommend submitting a new discussion in `Code Search and Navigation` addressing this as a separate topic; example: "GitHub enterprise accounts need support for excluding archived repositories in code search".
- in the body, maybe add a link to this discussion just for background information

It may seem redundant, but the new topic will help us track the community feedback more accurately and aid in collaboration.

**Comment by wyardley** [comment]
Is there a way to support `-is:archived` for issues / PRs, or would that have to be requested in a different place?

**Reply by LiteBrite82** [reply]
@wyardley 
Issues and PRs use a different search engine than code search. That would be a great discussion post for the `Projects and Issues` category. https://github.com/orgs/community/discussions/categories/projects-and-issues

**Comment by marcobiedermann** [comment]
You can simply click "Exclude archived" on the `/search` page, which results in `NOT is:archived`
<img width="288" alt="image" src="https://github.com/user-attachments/assets/5f4ed115-0bc6-4fa8-a5d9-5e10d76f5046" />

**Reply by bobpaul** [reply]
Only works for Code results.

1. Navigate to an org, ex github.com/mavlink
2. (Optional) On the list of repositories, look at the "Type" dropdown and cry a tear that there's still no "not archived" option here.
3. Click search at the top of the screen. The search box expands
4. Click "search in this organization"
5. Notice the search results have lots of results for Code, Repositories, Issues, Pull Requests, etc
6. On the left scroll down and find "Advanced -> Exclude archived". Click it.
7. Notice there's now 0 results for Repositories, Issues, and Pull requests, but the Code results do seem to exclude archived repos.
8. Repeat steps 1-5
9. At the top of the filter list, choose "repositories"
10. Scroll to the bottom of the filter list and notice "exclude archived" is not an option.
11. Click "archived". The search box now looks like `org:theOrgName archived:` and no repository results are found, even though there are archived repos for this org. (`is:archived` doesn't work for the repository results, either).

**Comment by jerryjvl** [comment]
It would be great for "Exclude archived" to be the default though, and searching in them an opt-in -- in an org with many archived repos it is really tedios having to re-apply this filter for every new search when it feels excluding code we have chosen to archive already sufficiently indicated our default intent to want to hide it(!?)

**Comment by asarkar** [comment]
Although this question is about excluding archived repos from search, a similar ask is to hide the archived repos when browsing the repositories (**Repositories** tab). This can be achieved by appending `?q=archived:false` to the URL as documented [here](https://docs.github.com/en/search-github/searching-on-github/searching-for-repositories#search-based-on-whether-a-repository-is-archived).

```
https://github.com/<user>?q=archived%3Afalse&tab=repositories
```

where `<user>` is a placeholder for the GitHub username whose repositories are being browsed.

**Reply by jeffshrager** [reply]
Ugly!

**Comment by whisller** [comment]
`-archived:true` that works for me, excludes ALL archived repositories from list of repositories.

**Reply by bitti** [reply]
> `-archived:true`

Wow, that instead of just `archived:false`? How do you came up with that? But I just tested `archived:false` and it also works for me. Was there a recent GHES release where this feature was included?

**Comment by drullo** [comment]
I guess I'll go ahead and state the obvious... nobody is EVER going to remember the syntax of some obscure/hidden feature or remember to refer back to this confusing thread with several variations of attempts.  This is such a prominent requirement for most people that it needs to just be a 3-state control in the search GUI.  We don't want to type additional search parameters and have it maybe work if we remembered the right syntax.  We want to type in our search terms and tick an option on/off to include or exclude archived repos.

Options: "Active Repos", "Archived Repos", "All Repos"
